File: app/src/main/java/mydebts/android/app/feature/events/EventsSubcomponent.java
package mydebts.android.app.feature.events;

import dagger.Subcomponent;
import mydebts.android.app.di.SingleIn;
import mydebts.android.app.di.SubcomponentBuilder;

@SingleIn(EventsFragment.class)
@Subcomponent
public interface EventsSubcomponent {

    void inject(EventsFragment fragment);

    @Subcomponent.Builder
    interface Builder extends SubcomponentBuilder {

        EventsSubcomponent build();
    }
}

File: app/src/main/java/mydebts/android/app/feature/events/EventsSubcomponent.kt
package mydebts.android.app.feature.events

import dagger.BindsInstance
import dagger.Subcomponent
import mydebts.android.app.di.SingleIn
import mydebts.android.app.di.SubcomponentBuilder

@SingleIn(EventsFragment::class)
@Subcomponent(modules = arrayOf(EventsModule::class))
interface EventsSubcomponent {

    fun inject(fragment: EventsFragment)

    @Subcomponent.Builder
    interface Builder : SubcomponentBuilder {

        @BindsInstance fun fragent(fragment: EventsFragment): Builder

        fun build(): EventsSubcomponent
    }
}
